gnark-prover/prover/combined_proving_system.go

```diff
@@ -16,23 +16,23 @@ type CombinedParameters struct {
 	NonInclusionParameters NonInclusionParameters
 }
 
-func (p *CombinedParameters) NumberOfUTXOs() uint32 {
+func (p *CombinedParameters) NumberOfUTXOs() int {
```

Comment:
why switch to int?

Reply:
(in case that it is unrelated to the single endpoint it would make reviewing a lot easier to do the switch in a different pr)